encoding/ssz/htrutils.go

```diff
@@ -56,22 +55,22 @@ func CheckpointRoot(hasher HashFn, checkpoint *ethpb.Checkpoint) ([32]byte, erro
 	return BitwiseMerkleize(hasher, fieldRoots, uint64(len(fieldRoots)), uint64(len(fieldRoots)))
 }
 
-// HistoricalRootsRoot computes the HashTreeRoot Merkleization of
-// a list of [32]byte historical block roots according to the Ethereum
-// Simple Serialize specification.
-func HistoricalRootsRoot(historicalRoots [][]byte) ([32]byte, error) {
-	result, err := BitwiseMerkleize(hash.CustomSHA256Hasher(), historicalRoots, uint64(len(historicalRoots)), params.BeaconConfig().HistoricalRootsLimit)
+// ByteArrayRootWithLimit computes the HashTreeRoot Merkleization of
+// a list of [32]byte roots according to the Ethereum Simple Serialize
+// specification.
+func ByteArrayRootWithLimit(roots [][]byte, limit uint64) ([32]byte, error) {
```

Comment:
Looks OK, but it is a little bit outside of the scope of a refactor PR.

Reply:
The reason was that this package had a dependency on the params package. 
The change was to reduce the likelihood of a circular dependency in tests and make this library more generic.